craft_parts/utils/maven/common.py

```diff
@@ -147,46 +148,43 @@ def update_pom(
     :param self_contained: Whether or not to patch version numbers with what is
         actually available.
     """
-    pom_xml = part_info.part_build_subdir / "pom.xml"
-
-    if not pom_xml.is_file():
-        raise MavenXMLError("'pom.xml' does not exist")
-
-    tree = ET.parse(pom_xml)  # noqa: S314, unsafe parsing with xml
-
-    project = tree.getroot()
-    namespace = re.search("{(.*)}", project.tag)
-
-    # Older pom files may not contain a namespace
-    namespaces = {"": namespace.group(1)} if namespace else {}
-    for prefix, uri in namespaces.items():
-        ET.register_namespace(prefix, uri)
+    existing = _get_existing_artifacts(part_info)
+    poms = _get_poms(part_info, existing)
 
-    if deploy_to is not None:
-        # Add a distributionManagement element, to tell "maven deploy" to deploy the
-        # artifacts (jars, poms, etc) to the export dir.
-        distribution_dir = deploy_to / "maven-use"
-        distribution_element = ET.fromstring(  # noqa: S314, unsafe parsing with xml
-            DISTRIBUTION_REPO_TEMPLATE.format(repo_uri=distribution_dir.as_uri())
-        )
+    # We should at least have a single pom file
+    if not poms or not poms[0].is_file():
```

Comment:
When would `poms[0]` not be a file? Following up on my other comment, I feel like this list should only contain existing poms